src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -343,9 +413,25 @@ void setNotifier(Notifier* n) {
 */
 DBClientBase* getMongoConnection(void)
 {
-    return connection;
+#ifdef UNIT_TEST
+  return connection;
+#else
+  return mongoPoolConnectionGet();
+#endif
 }
 
+DBClientBase* mongoInitialConnectionGetForUnitTest(void)
+{
+  return mongoPoolConnectionGet();
+}
+
+void releaseMongoConnection(DBClientBase* connection)
```

Comment:
The function to GET a connection is called getMongoConnection (and it calls internally mongoPoolConnectionGet).
The function to RELEASE that connection needs a corresponding name, such as releaseMongoConnection. 
Right now, this function only calls mongoPoolConnectionRelease, but this might change in the future. 
I see no reason to change anything here (even though it is true that we'd gain some execution time calling mongoPoolConnectionRelease directly instead of calling releaseMongoConnection that calls mongoPoolConnectionRelease.

The code was already **full** of calls to getMongoCollection, and we can't have code like this:

    connection = getMongoCollection();
    useConnection();
    mongoPoolConnectionRelease();

This would be hard to understand.

I really think we should keep it as is. I see no better solution (unless we start to hunt nanoseconds in run-time).